filestorage.go

```diff
@@ -199,27 +214,29 @@ createLock:
 			}
 
 			// if lock is not stale, wait upon it
-			return fw, nil
+			fw.Wait()
+			return nil
 		}
 
 		// otherwise, this was some unexpected error
-		return nil, err
+		return err
 	}
 	lf.Close()
 
-	// looks like we get the lock
-	fw.wg.Add(1)
-	fileStorageNameLocks[key] = fw
-
-	return nil, nil
+	// cool, we got the lock right away
+	return nil
 }
 
 // Unlock releases the lock for name.
 func (fs FileStorage) Unlock(key string) error {
 	fileStorageNameLocksMu.Lock()
 	defer fileStorageNameLocksMu.Unlock()
 
-	fw, ok := fileStorageNameLocks[key]
+	if fs.fileStorageNameLocks == nil {
```

Comment:
The map is never *written to* so it doesn't need to be initialized.

Reply:
Good point!